tensorflow/cc/framework/gradients_test.cc

```diff
@@ -431,6 +431,39 @@ TEST_F(GradientsTest, UnreachableEdgeGradTwoOutputs) {
       outputs[0], test::AsTensor<double>({17.5, 24.7, 26.8}, {3, 1}), 1e-5);
 }
 
+TEST_F(GradientsTest, UnreachableInput) {
+  auto x = Variable(scope_test_, {2, 3}, DT_DOUBLE);
+  auto x_const = Const(scope_test_, {{1.0, 2.0, 3.0}, {4.0, 5.0, 6.0}});
+  auto x_assign = Assign(scope_test_, x, x_const);
```

Comment:
Do you need to use the Variable-Const-Assign pattern here, or can you just use Const or Placeholder nodes? I realize this is very similar to the discussion we had in the last PR, but I think in this case it'll be easier to understand and verify what the test is actually testing if you use simple inputs.

Reply:
Verified.